modin/tests/pandas/utils.py

```diff
@@ -1086,14 +1086,25 @@ def eval_io_from_str(csv_str: str, unique_filename: str, **kwargs):
     )
 
 
-def create_test_dfs(*args, **kwargs) -> tuple[pd.DataFrame, pandas.DataFrame]:
-    post_fn = kwargs.pop("post_fn", lambda df: df)
+def create_test_dfs(
+    *args, post_fn=None, backend=None, **kwargs
+) -> tuple[pd.DataFrame, pandas.DataFrame]:
+    if post_fn is None:
+        post_fn = lambda df: (  # noqa: E731
+            df.convert_dtypes(dtype_backend=backend) if backend is not None else df
+        )
+    elif backend is not None:
```

Comment:
```suggestion
    elif backend is not None:
```
```suggestion
    elif backend is not None and post_fn is not None:
```
We use post_fn below. If it is None, we could get a failure.

Reply:
We have a separate `if post_fn is None:` branch before. post_fn` can't be None here.